daemon/api_download.go

```diff
@@ -54,23 +55,34 @@ func postSnapDownload(c *Command, r *http.Request, user *auth.UserState) Respons
 		return BadRequest("download operation requires one snap name")
 	}
 
-	return streamOneSnap(c, user, action.SnapName)
+	return streamOneSnap(c, user, action)
 }
 
-func streamOneSnap(c *Command, user *auth.UserState, snapName string) Response {
-	info, err := getStore(c).SnapInfo(context.TODO(), store.SnapSpec{Name: snapName}, user)
+func streamOneSnap(c *Command, user *auth.UserState, opts snapDownloadAction) Response {
+	actions := []*store.SnapAction{{
+		Action:       "download",
+		InstanceName: opts.SnapName,
+		Revision:     opts.Revision,
+		CohortKey:    opts.CohortKey,
+		Channel:      opts.Channel,
+	}}
+	snaps, err := getStore(c).SnapAction(context.TODO(), nil, actions, user, nil)
 	if err != nil {
-		return SnapNotFound(snapName, err)
+		return SnapNotFound(opts.SnapName, err)
```

Comment:
we probably want to use:

errToResponse(err, []string{action.snapName}, InternalError, "cannot download: %v")

here